src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,345 @@
+    }
+
+    public new void ErasePreset(Color color)
+    {
+        OnPresetDeleted(presets.First(p => p.Color == color));
+    }
+
+    private void UpdateTooltips()
+    {
+        pickerButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_PICK_COLOR");
+        addPresetButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_ADD_PRESET");
+        hsvCheckButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_HSV_BUTTON_TOOLTIP");
+        rawCheckButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_RAW_BUTTON_TOOLTIP");
+
+        if (HsvMode)
+        {
+            sliderROrH.HintTooltip = TranslationServer.Translate("COLOR_PICKER_H_TOOLTIP");
+            sliderGOrS.HintTooltip = TranslationServer.Translate("COLOR_PICKER_S_TOOLTIP");
+            sliderBOrV.HintTooltip = TranslationServer.Translate("COLOR_PICKER_V_TOOLTIP");
+        }
+        else
+        {
+            sliderROrH.HintTooltip = TranslationServer.Translate("COLOR_PICKER_R_TOOLTIP");
+            sliderGOrS.HintTooltip = TranslationServer.Translate("COLOR_PICKER_G_TOOLTIP");
+            sliderBOrV.HintTooltip = TranslationServer.Translate("COLOR_PICKER_B_TOOLTIP");
+        }
+
+        sliderA.HintTooltip = TranslationServer.Translate("COLOR_PICKER_A_TOOLTIP");
+    }
+
+    private void UpdateButtonsState()
+    {
+        if (hsvCheckButton == null)
+            return;
+
+        hsvCheckButton.Pressed = HsvMode;
+        rawCheckButton.Pressed = RawMode;
+
+        hsvCheckButton.Disabled = !hsvButtonEnabled || RawMode;
+        rawCheckButton.Disabled = !rawButtonEnabled || HsvMode;
+    }
+
+    private void OnAddPresetButtonPressed()
+    {
+        AddPreset(Color);
+    }
+
+    private void OnPresetSelected(Color color)
+    {
+        Color = color;
+        EmitSignal("color_changed", Color);
+    }
+
+    private void OnPresetDeleted(TweakedColorPickerPreset preset)
+    {
+        presets.Remove(preset);
+        presetsContainer.RemoveChild(preset);
+        base.ErasePreset(preset.Color);
+        preset.QueueFree();
+    }
+
+    private void OnHSVButtonToggled(bool isOn)
+    {
+        HsvMode = isOn;
+    }
+
+    private void OnRawButtonToggled(bool isOn)
+    {
+        RawMode = isOn;
+    }
+
+    private void OnColorChanged(Color color)
+    {
+        hexColorEdit.Text = color.a8 == 255 ? color.ToHtml(false) : color.ToHtml(true);
+    }
+
+    /// <summary>
+    ///   Called when (keyboard) entered in hexColorEdit.
+    ///   Set Color when text is valid; reset if not.
+    ///   Because text_entered signal requires a string parameter but focus_exited zero, wrote lite this.
+    /// </summary>
+    /// <param name="color">Current hexColorEditor text</param>
+    private void OnHexColorChanged(string color = null)
+    {
+        color ??= hexColorEdit.Text;
+        if (color.IsValidHtmlColor())
+            Color = new Color(color);
+        else
+            hexColorEdit.Text = Color.ToHtml();
+    }
```

Comment:
We usually use brackets for if else blocks.
```csharp
color ??= hexColorEdit.Text;

if (color.IsValidHtmlColor())
{
    Color = new Color(color);
}
else
{
    hexColorEdit.Text = Color.ToHtml();
}
```

Reply:
Even though Godot requires many uses of Colour, our variables should be named colour as per the style guide saying to use british spelling.